File: codeBase/ACES_Coders/v8.0/pre-coders/Lexicographically-smallest-game/Solution.py
'''
    This code was written by UOP_hashInclude during ACES Precoders v8 2020
    nuwanjaliyagoda@eng.pdn.ac.lk   imeshuek@eng.pdn.ac.lk  dinelkadilshani95@gmail.com

'''
n=int(input())
p=[]
for i in range (n):
    word=input().strip()
    p+=[word]


for i in p:
    w=""
    for j in i:
        if w=="":
            w=j
        elif w[0]>=j:
            w=j+w
        else:
            w=w+j
    print(w)
        
    

File: codeBase/ACES_Coders/v8.0/pre-coders/Lexicographically-smallest-game/Solution.cpp
/*
	This code was written by UOP_JPAC during ACES Precoders v8 2020
	pubuduudara7@gmail.com	imsuneth@gmail.com	pasan96tennakoon@gmail.com


*/
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;


int main() {
    int t;
    cin >> t;
    for(int k=0;k<t;++k){
        string s;
        cin >> s;
        string result;
        result += s[0];
        for(int i=1;i<s.length();++i)
        {
            if(s[i] > result[0]){
                result = result + s[i];
            }   
            else{
                result = s[i] + result;
            }
        }
        cout << result << endl;
    }
    return 0;
}